pkg/skaffold/schema/v1alpha2/config.go

```diff
@@ -238,17 +241,31 @@ func (c *SkaffoldConfig) setDefaultWorkspaces() {
 	}
 }
 
-func (c *SkaffoldConfig) expandKanikoSecretPath() error {
-	if c.Build.KanikoBuild == nil || c.Build.KanikoBuild.PullSecret == "" {
-		return nil
+func (c *SkaffoldConfig) setDefaultKanikoNamespace() {
+	if c.Build.KanikoBuild == nil {
+		return
+	}
+	if c.Build.KanikoBuild.Namespace == "" {
+		c.Build.KanikoBuild.Namespace = "default"
```

Comment:
This should be resolved from the kubernetes client. The default namespace will be different if you configure your context with a different namespace.

Reply:
I switched this up to just pass "" by default, which seems to be what we do everywhere else: https://github.com/GoogleContainerTools/skaffold/blob/master/pkg/skaffold/kubernetes/log.go#L71

Reply:
Thats a little different - "" is all namespaces